acra/security-controls/sql-firewall/_index.md

```diff
@@ -2,3 +2,236 @@
+
+
+
+
+### Prepared statements
+
+The root of the SQL injection problem is that the code and the data (withing SQL query) can get mixed up and be mistaken for each other. SQL prepared statements help to separate code execution from the data itself.
+
+AcraCensor has limited support of filtering prepared statements. AcraCensor doesn't apply allow/deny rules for SQL requests with prepared statements. However, it's possible to use query_capture/query_ignore to filter the whole statement when necessary (through comparing the incoming SQL char-by-char with the SQL from the config file).
+
+
+### Ignoring specific queries
+
+It might be useful to skip appliance of allowlist/denylist rules to particular queries, for example, the SQL queries on starting up the database. If the query is ignored, AcraCensor won't apply any rules to it (so basically AcraCensor will pass this query to the database).
+
+```
+- handler: query_ignore
+  queries:
+    # mysql queries AcraCensor will ignore
+    - ROLLBACK
+    - SET AUTOCOMMIT = 0
+    - SHOW VARIABLES LIKE 'sql_mode'
+    - show collation where `Charset` = 'utf8' and `Collation` = 'utf8_bin'
+```
+
+To ignore some queries, put them to the `query_ignore` handler. AcraCensor compares each input query with queries from the `query_ignore` list. If there is a match, the query is allowed through and no allow/deny handlers are applied.
+
+We suggest looking into [tests/acra-censor_configs](https://github.com/cossacklabs/acra/tree/master/tests/acra-censor_configs) to see which PostgreSQL and MySQL queries we've added to the ignore list for running integration tests.
+
+### Unparseable queries
+
+Sometimes AcraCensor can't parse some particular SQL queries. It's potentially dangerous to send such queries to the database as they might contain SQL injections. So for security reasons, AcraCensor blocks "unparseable" queries by default.
+
+However, some of these queries might be totally legit, so users should have a way to allow them. To remedy this, we've added a special `ignore_parse_error` field into the configuration.
+
+```
+ignore_parse_error: false
+```
+
+By default, `ignore_parse_error` is false in AcraCensor (check [configs/acra-censor.example.yaml](https://github.com/cossacklabs/acra/blob/master/configs/acra-censor.example.yaml)). Setting it to `true` will push AcraCensor to ignore "unparseable" queries and send them to the database — allowing developers to configure AcraCensor rules during the development and integration stages.
+
+In a production environment, security-oriented users keep `ignore_parse_error` disabled, but add such “unparseable but legit” queries to the `query_ignore` handler manually, resulting in a well-defined and secure firewall configuration. The users encountering parsing errors can send us their GitHub Issue or PR and we'll look into it and try to help.
+
+
+### Logging and masking queries
+
+#### Continuous logging into syslog
+
```

Comment:
I don't get why this subsection header is necessary here. Also, what does syslog have to do with this and why this is the only place where it's mentioned?

```suggestion
### Logging and masking queries
```

Reply:
removed